### PR TITLE
handling individual higlight msg

### PR DESCRIPTION
**Pull request by xxjunxx**

### CODE REVIEW

File: Source/USemLog/Classes/Owl/SLOwlExperiment.h [hunk 1]
```diff
@@ -28,13 +27,13 @@ struct FSLOwlExperiment : FSLOwlDoc
 	TArray<FSLOwlNode> TimepointIndividuals;
 
 	// Set of registered timepoints (in order to avoid multiple individual declaration)
-	TArray<float> RegisteredTimepoints;
+	TSet<float> RegisteredTimepoints;
```
Comment (andreihaidu) [inline Source/USemLog/Classes/Owl/SLOwlExperiment.h:30]: this should stay array

File: Source/USemLog/Classes/Owl/SLOwlExperiment.h [hunk 2]
```diff
@@ -28,13 +27,13 @@ struct FSLOwlExperiment : FSLOwlDoc
...
 
 	// Array of object individuals
 	TArray<FSLOwlNode> ObjectIndividuals;
 
 	// Set of registered objects (in order to avoid multiple individual declaration)
-	TSet<USLBaseIndividual*> RegisteredObjects;
+	TSet<UObject*> RegisteredObjects;
```
Comment (andreihaidu) [inline Source/USemLog/Classes/Owl/SLOwlExperiment.h:36]: should stay USLBaseIndividual

File: Source/USemLog/Classes/Owl/SLOwlExperiment.h [hunk 3]
```diff
@@ -53,60 +52,46 @@ struct FSLOwlExperiment : FSLOwlDoc
 	// Destructor
 	~FSLOwlExperiment() {}
 
-	// Add timepoint individual value
-	void RegisterTimepoint(const float Timepoint)
+	// Add timepoint individual
+	bool AddTimepointIndividual(const float Timepoint, const FSLOwlNode& InOwlNode)
```
Comment (andreihaidu) [inline Source/USemLog/Classes/Owl/SLOwlExperiment.h:56]: this should not change

File: Source/USemLog/Private/Editor/SLSemanticMapWriter.cpp [hunk 4]
```diff
@@ -189,19 +189,18 @@ void FSLSemanticMapWriter::AddObjectIndividual(TSharedPtr<FSLOwlSemanticMap> InS
...
-		//	if (UStaticMeshComponent* SMC = ActAsSMA->GetStaticMeshComponent())
-		//	{
-		//		if (UStaticMesh* SM = SMC->GetStaticMesh())
-		//		{
-		//			//ObjIndividual.AddChildNode(FSLOwlSemanticMapStatics::CreatePathToCadModelProperty(GetPathToCadModelLambda(SM)));
-		//			ObjIndividual.AddChildNode(FSLOwlSemanticMapStatics::CreatePathToCadModelProperty(InClass));
-		//		}
-		//	}
-		//}
+		// If static mesh, add pathToCadModel property
```
Comment (andreihaidu) [inline Source/USemLog/Private/Editor/SLSemanticMapWriter.cpp:192]: file should stay the same

File: Source/USemLog/Private/Events/SLContactEvent.cpp [hunk 5]
```diff
@@ -50,10 +50,14 @@ void FSLContactEvent::AddToOwlDoc(FSLOwlDoc* OutDoc)
 	// we cannot use the safer dynamic_cast because RTTI is not enabled by default
 	// if (FOwlEvents* EventsDoc = dynamic_cast<FOwlEvents*>(OutDoc))
 	FSLOwlExperiment* EventsDoc = static_cast<FSLOwlExperiment*>(OutDoc);
-	EventsDoc->RegisterTimepoint(StartTime);
-	EventsDoc->RegisterTimepoint(EndTime);
-	EventsDoc->RegisterObject(Individual1);
-	EventsDoc->RegisterObject(Individual2);
+	EventsDoc->AddTimepointIndividual(
```
Comment (andreihaidu) [inline Source/USemLog/Private/Events/SLContactEvent.cpp:53]: leave with RegisterTimepoint

**Comment by andreihaidu** [comment]
@xxjunxx there are still too many changes, only comment out the 'EpisodeId' related parts. I can also take a look myself, and then you just sync your changes.

**Comment by xxjunxx** [comment]
how about just close the pull request and I will try just comment 'EpisodeId'

**Comment by andreihaidu** [comment]
@xxjunxx yes, it would be the quickest approach

**Pull request closed without merging**